crypto/hasher/hasher.go

```diff
@@ -0,0 +1,53 @@
+package hasher
+
+import (
+	"fmt"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+)
+
+// Hash represents the output of the used hash function.
+type Hash [crypto.DefaultHashSizeByte]byte
+
+// PADHasher provides hash functions for the PAD implementations.
+type PADHasher interface {
+	// ID returns the name of the cryptographic hash function.
+	ID() string
+	// Size returns the size of the hash output in bytes.
+	Size() int
+	// Digest hashes all passed byte slices. The passed slices won't be mutated.
+	Digest(ms ...[]byte) []byte
+	treeHasher
+}
+
+// treeHasher provides hash functions for tree implementations.
+type treeHasher interface {
```

Comment:
On first sight, it's a bit confusing that the `treeHasher` is unexported but embedded in the exported `PADHasher` interface. Maybe there is a good reason which I don't see yet? OK, the 'tree-hashing logic' is separate from the rest (ID, Size, ...) but  if one wants to implement a different `PADHasher`, one needs to find the private interface `treeHasher`. I think it makes sense to collapse both and a public interface should be self-contained (or embed other public interfaces if there are really good reasons). LMK what you think.

Reply:
I don't remember why I didn't publish `treeHasher` interface but I can't find any reasons for not doing that.

> I think it makes sense to collapse both and a public interface should be self-contained (or embed other public interfaces if there are really good reasons).

Can you elaborate?

Reply:
I think this should be one interface instead of two. At least both should be public if there is a good reason for having two ifaces (which I don't see).

Reply:
Done.